### ISSUE TITLE
Conditional loading of Swiper: CORS issue on multidomain setup

### ISSUE THREAD

**Issue by nicmare** [elementor/elementor#13701]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] The issue still exists against the latest stable version of Elementor.

## Description

Conditional loading of swiper works fine for primary domain pages. But when using polylang or wpml with separate domain for each language, the swiper can't be loaded because of cross domain origin. To fix similar problems in the past, i used a php snippet to change the paths like:
```
add_filter( 'script_loader_src', 'wpse47206_src' );
add_filter( 'style_loader_src', 'wpse47206_src' );
function wpse47206_src( $url )
{
    if (strpos($_SERVER['SERVER_NAME'],'domain.io') !== false) {
        return str_replace('domain.de', 'domain.io', $url);
    }
    return $url;
}
```
This is a simple fix which worked out well. But with the new conditional swiper loading, it seems to work differently. Somehow loaded through frontend.js – therefore the filter does not work and resulting in:

![image](https://user-images.githubusercontent.com/898582/105967765-864cdb00-6086-11eb-96a3-31e27c89f026.png)

## Steps to reproduce

1. activate Improved Asset Loading
2. add polylang or wpml and a secondary domain
3. insert a swiper widget on the secondary domain page

## Isolating the problem

- [x] This bug happens with only Elementor plugin active (and Elementor Pro).
- [x] This bug happens with a default WordPress theme active.
- [x] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>
```
					
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: MariaDB Server v5.5.61-MariaDB-38.13
	PHP Version: 7.3.26
	PHP Max Input Vars: 1000
	PHP Max Post Size: 16M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.5.3
	Site URL: https://softgarden.de
	Home URL: https://softgarden.de
	WP Multisite: No
	Max Upload Size: 11 GB
	Memory limit: 256M
	Permalink Structure: /blog/%postname%/
	Language: en-US
	Timezone: 2
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor Child
	Version: 1.0.1
	Author: Elementor Team
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 2.3.1
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 11_1_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/88.0.4324.96 Safari/537.36

== Active Plugins ==
	Advanced Custom Fields PRO
		Version: 5.9.4
		Author: Elliot Condon

	Akismet Anti-Spam
		Version: 4.1.8
		Author: Automattic

	DynamicConditions
		Version: 1.4.6
		Author: RTO GmbH

	Elementor
		Version: 3.1.0
		Author: Elementor.com

	Elementor Pro
		Version: 3.0.10
		Author: Elementor.com

	Elementor Softgarden
		Version: 1.0.0
		Author: Peak Ace AG

	Elementor Widgets NB
		Version: 1.0.0
		Author: NB

	Enable Media Replace
		Version: 3.5.0
		Author: ShortPixel

	Extras for Elementor
		Version: 2.2.44
		Author: Namogo

	Formidable Forms
		Version: 4.09.04
		Author: Strategy11

	Formidable Forms Pro
		Version: 4.09.04
		Author: Strategy11

	Functions NB
		Version: 1.0
		Author: Nic Bugus

	Happy Elementor Addons
		Version: 2.19.0
		Author: weDevs

	Happy Elementor Addons Pro
		Version: 1.13.1
		Author: weDevs

	Header Footer Code Manager
		Version: 1.1.8
		Author: 99robots

	Polylang Connect for Elementor
		Version: 1.0.0
		Author: David Decker - DECKERWEB

	Polylang Pro
		Version: 2.9.1
		Author: WP SYNTEX

	Post Type Archive Descriptions
		Version: 1.2.0
		Author: Mark Root-Wiley, MRW Web Design, NonprofitWP.org

	Really Simple SSL
		Version: 4.0.6
		Author: Really Simple Plugins

	Redirection
		Version: 4.9.2
		Author: John Godley

	Safe SVG
		Version: 1.9.9
		Author: Daryll Doyle

	Tuxedo Big File Uploads
		Version: 1.2
		Author: Trevor Anderson

	WP Migrate DB Pro
		Version: 1.9.13
		Author: Delicious Brains

	Yoast SEO
		Version: 15.7
		Author: Team Yoast


== Must-Use Plugins ==
	WP Migrate DB Pro Compatibility
		Version: 1.2
		Author: Delicious Brains


== Elementor - Compatibility Tag ==
	
	Elementor Pro: header_not_exists
	Elementor Softgarden: header_not_exists
	Elementor Widgets NB: header_not_exists
	Extras for Elementor: header_not_exists
	Happy Elementor Addons: incompatible
	Happy Elementor Addons Pro: header_not_exists
	Polylang Connect for Elementor: header_not_exists

== Elements Usage ==
	
	archive : 11
		archive-posts : 1
		button : 1
		button-group : 2
		ee-search-form : 2
		heading : 23
		posts : 8
		search-form : 3
		spacer : 11
		text-editor : 8
		theme-archive-title : 2
	footer : 5
		button : 7
		divider : 4
		form : 4
		heading : 17
		html : 1
		image : 21
		nav-menu : 10
		posts : 3
		social-icons : 5
		spacer : 3
		text-editor : 21
		wpml-language-switcher : 1
	header : 11
		button : 10
		icon : 7
		image : 1
		jet-mega-menu : 1
		main-header : 6
		main-language-switcher : 1
		nav-menu : 5
		polylang-language-switcher : 5
		star-rating : 3
		theme-site-logo : 1
	page : 25
		button : 29
		divider : 5
		gallery : 15
		heading : 155
		html : 8
		icon-list : 28
		image : 113
		image-box : 2
		menu-anchor : 4
		nav-menu : 1
		posts : 1
		price-table : 19
		recruitingcalculator : 2
		spacer : 322
		tabs : 4
		testimonial-carousel : 4
		text-editor : 205
		tooltip-list : 3
		video : 5
	popup : 12
		button : 6
		heading : 70
		image : 3
		recruitingcalculator : 2
		recruitingcalculatorresults : 2
		spacer : 15
		text-editor : 8
	section : 36
		button : 20
		button-group : 1
		divider : 12
		ee-search-form : 1
		gallery : 3
		global : 3
		heading : 59
		html : 3
		hubspotform : 1
		icon : 38
		icon-list : 2
		image : 16
		recruitingcalculator : 2
		recruitingcalculatorresults : 1
		slides : 3
		spacer : 46
		star-rating : 15
		tabs : 1
		text-editor : 42
		tooltip-list : 1
	single : 22
		author-box : 2
		button : 2
		button-group : 1
		ee-search-form : 1
		gallery : 2
		heading : 47
		html : 3
		hubspotform : 1
		icon-list : 4
		image : 14
		post-comments : 1
		post-info : 4
		posts : 2
		share-buttons : 1
		shortcode : 1
		spacer : 74
		table-of-contents : 1
		testimonial-carousel : 2
		text-editor : 45
		theme-post-content : 6
		theme-post-excerpt : 1
		theme-post-title : 5
	widget : 0
		author-box : 1
		gallery : 369
		main-filter : 11
		nav-menu : 1
	wp-post : 55
		button : 1
		call-to-action : 74
		counter : 3
		heading : 248
		html : 1
		image : 17
		menu-anchor : 2
		progress : 1
		read-more : 10
		recruitingcalculator : 1
		recruitingcalculatorresults : 1
		spacer : 29
		text-editor : 416
		video : 4
	wp-page : 222
		alert : 3
		animated-headline : 17
		archive-posts : 1
		blockquote : 24
		button : 179
		button-group : 1
		call-to-action : 34
		circle-progress : 1
		countdown : 2
		counter : 19
		divider : 33
		ee-inline-svg : -5
		ee-search-form : 1
		flip-box : 2
		form : 3
		gallery : -12
		global : 3
		ha-animated-text : 3
		ha-fun-factor : 14
		ha-image-grid : 2
		ha-logo-carousel : 4
		ha-logo-grid : 9
		ha-post-grid : 4
		ha-post-tiles : 3
		ha-skills : 5
		ha-step-flow : 16
		ha-testimonial-carousel : 1
		ha-timeline : 4
		ha-toggle : 1
		heading : 578
		html : -25
		html5-video : 9
		hubspotform : 57
		icon : 189
		icon-box : 31
		icon-list : 5
		image : 65
		image-box : 10
		image-carousel : 4
		inline-editing-example : 1
		main-download : 1
		main-navigation : 1
		main-tooltip-col : 2
		media-carousel : 5
		menu-anchor : 54
		pardotmarketing-form-handler : 3
		portfolio : 1
		posts : 17
		price-table : 5
		progress : 1
		recruitingcalculator : 7
		recruitingcalculatorresults : 6
		slides : 17
		spacer : -71
		star-rating : 44
		testimonial-carousel : 4
		text-editor : 794
		timeline : 2
		toggle : 7
		tooltip-list : 1
		video : 15
		wp-widget-pages : 3
		wp-widget-text : 3
	single-post : 2
		author-box : 1
		blockquote : 1
		button : 3
		button-group : 1
		ee-search-form : 1
		form : 1
		heading : 22
		html : 1
		image-gallery : 1
		portfolio : 1
		post-comments : 1
		post-info : 1
		posts : 2
		share-buttons : 1
		shortcode : 1
		spacer : 2
		table-of-contents : 1
		theme-post-content : 2
		theme-post-excerpt : 1
		theme-post-featured-image : 2
		theme-post-title : 2
	search-results : 1
		archive-posts : 1
		heading : 1
		search-form : 1
		spacer : 2


== Log ==
	
JS: showing 20 of 22JS: 2021-01-18 09:51:13 [error X 5][https://softgarden.de/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view281\") has already been destroyed and cannot be used. 
JS: 2021-01-18 13:27:30 [error X 1][https://softgarden.de/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.0.16:2:131641] Cannot read property \'isEditable\' of undefined 
JS: 2021-01-18 16:41:02 [error X 126][https://softgarden.de/wp-content/plugins/elementor/assets/js/editor-document.min.js?ver=3.0.16:2:6232] container or containers are required. 
JS: 2021-01-18 16:54:23 [error X 4][https://softgarden.de/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view251\") has already been destroyed and cannot be used. 
JS: 2021-01-19 10:32:34 [error X 4][https://softgarden.de/wp-content/plugins/elementor/assets/lib/backbone/backbone.marionette.min.js?ver=2.4.5.e1:24:19952] View (cid: \"view227\") has already been destroyed and cannot be used. 
JS: 2021-01-19 10:49:29 [error X 1][https://softgarden.de/wp-admin/post.php?post=64045&action=elementor#:3:13] Unexpected token \'do\' 
JS: 2021-01-19 10:49:31 [error X 1][https://softgarden.de/wp-admin/post.php?post=64045&action=elementor#:3:16] Unexpected token \'=\' 
JS: 2021-01-19 10:49:35 [error X 3][https://softgarden.de/wp-admin/post.php?post=64045&action=elementor#:5:9] Unexpected token \'if\' 
JS: 2021-01-19 10:49:36 [error X 6][https://softgarden.de/wp-admin/post.php?post=64045&action=elementor#:4:15] Missing initializer in const declaration 
JS: 2021-01-19 10:49:37 [error X 3][https://softgarden.de/wp-admin/post.php?post=64045&action=elementor#:4:15] Identifier \'element\' has already been declared 
JS: 2021-01-19 10:49:43 [error X 5][https://softgarden.de/wp-admin/post.php?post=64045&action=elementor#:5:27] Unexpected token \'{\' 
JS: 2021-01-19 10:50:30 [error X 1][https://softgarden.de/wp-admin/post.php?post=64045&action=elementor#:5:21] Unexpected token \'const\' 
JS: 2021-01-19 10:52:26 [error X 3][https://softgarden.de/wp-admin/post.php?post=64045&action=elementor#:5:37] Unexpected token \']\' 
JS: 2021-01-19 10:54:16 [error X 4][https://softgarden.de/wp-admin/post.php?post=64045&action=elementor#:8:9] missing ) after argument list 
JS: 2021-01-19 12:16:14 [error X 2][https://softgarden.de/wp-admin/post.php?post=64045&action=elementor#:4:33] Unexpected token \';\' 
JS: 2021-01-19 15:34:22 [error X 5][https://softgarden.io/product/headhunter-module-draft/:140:51] Cannot read property \'replace\' of undefined 
JS: 2021-01-22 13:33:49 [error X 3][https://softgarden.de/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.0.16:2:230493] Blocked a frame with origin \"https://softgarden.de\" from accessing a cross-origin frame. Protocols, domains, and ports must match. 
JS: 2021-01-22 13:51:30 [error X 2][https://softgarden.io/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.0.16:2:204707] undefined is not an object (evaluating \'elementorFrontend.config.elements.data\') 
JS: 2021-01-26 09:02:23 [error X 12][https://softgarden.de/wp-content/plugins/elementor-extras/assets/lib/hotips/hotips.min.js?ver=1.1.0:1:6862] TweenMax is not defined 
JS: 2021-01-26 12:51:13 [error X 6][https://softgarden.de/wp-content/plugins/elementor/assets/js/editor.min.js?ver=3.1.0:2:220731] Blocked a frame with origin \"https://softgarden.de\" from accessing a cross-origin frame. 

PHP: showing 6 of 6PHP: 2021-01-19 03:40:52 [notice X 5][/var/www/vhosts/softgarden.io/softgarden.io/wp-content/plugins/elementor-pro/core/utils.php::217] Trying to get property 'post_author' of non-object [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2021-01-19 12:27:43 [notice X 721][/var/www/vhosts/softgarden.io/softgarden.io/wp-content/plugins/elementor-pro/modules/dynamic-tags/acf/tags/acf-image.php::56] Undefined index: preview_size [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2021-01-19 13:27:42 [notice X 9][/var/www/vhosts/softgarden.io/softgarden.io/wp-content/plugins/elementor/core/common/modules/ajax/module.php::171] Undefined index: data [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2021-01-22 15:33:53 [notice X 145][/var/www/vhosts/softgarden.io/softgarden.io/wp-content/plugins/elementor/includes/template-library/sources/local.php::588] Trying to get property 'display_name' of non-object [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2021-01-25 12:30:55 [warning X 1][/var/www/vhosts/softgarden.io/softgarden.io/wp-content/plugins/elementor/core/files/assets/svg/svg-handler.php::101] file_get_contents(/var/www/vhosts/softgarden.io/softgarden.io/wp-content/uploads/2018/12/Element-33.svg): failed to open stream: No such file or directory [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2021-01-25 14:15:40 [warning X 2][/var/www/vhosts/softgarden.io/softgarden.io/wp-content/plugins/elementor/core/files/assets/svg/svg-handler.php::101] file_get_contents(/var/www/vhosts/softgarden.io/softgarden.io/wp-content/uploads/2018/09/logo-color-finanzcheck.svg): failed to open stream: No such file or directory [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

Log: showing 14 of 142021-01-26 14:47:12 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.0.16',
  'to' => '3.1.0',
)]
2021-01-26 14:47:12 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.0.9',
  'to' => '3.0.10',
)]
2021-01-26 14:47:20 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.0.16',
  'to' => '3.1.0',
)]
2021-01-26 14:47:20 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.0.9',
  'to' => '3.0.10',
)]
2021-01-26 14:47:21 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.0.16',
  'to' => '3.1.0',
)]
2021-01-26 14:47:21 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.0.9',
  'to' => '3.0.10',
)]
2021-01-26 14:47:28 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.0.16',
  'to' => '3.1.0',
)]
2021-01-26 14:47:28 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.0.9',
  'to' => '3.0.10',
)]
2021-01-26 14:47:30 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.0.16',
  'to' => '3.1.0',
)]
2021-01-26 14:47:30 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.0.9',
  'to' => '3.0.10',
)]
2021-01-26 14:47:31 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.0.16',
  'to' => '3.1.0',
)]
2021-01-26 14:47:31 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.0.9',
  'to' => '3.0.10',
)]
2021-01-26 14:47:34 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.0.16',
  'to' => '3.1.0',
)]
2021-01-26 14:47:34 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.0.9',
  'to' => '3.0.10',
)]


				
```
</details>

**Comment by DorShahar** [comment]
Hi @nicmare .
Thanks for letting us know.
We will investigate this issue.

**Comment by DorShahar** [comment]
Hi,
We are working on a fix.
Thanks for informing us.

**Comment by shilo-ey** [comment]
This issue has been resolved in Elementor v3.1.2

Feel free to update

Thanks!
